### PR TITLE
feat(nitro): improve externals

### LINKED ISSUE #1670: import externals with full path for nitro dev
In development, we generate `.nuxt/nitro/index.mjs` with externals implicitly pointing to `node_modules` (like `import destr from 'destr';`) . This makes two major issues:
- Node.js has different search path for `node_modules` than the one configured for rollup and as a result hoisting issues
- With pnp (#581) we need explicit dependencies while imports like `h3` are actually not a dependency of the project

### PR DESCRIPTION
**Pull request by pi0**
### 🔗 Linked issue

resolves #1670

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

- [x] Resolve externals with full path
- [ ] Use externality

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/nitro/src/rollup/plugins/externals.ts [hunk 1]
```diff
@@ -60,8 +63,16 @@ export function externals (opts: NodeExternalsOptions): Plugin {
         }
       }
 
+      const resolved = await this.resolve(originalId, importer, { ...options, skipSelf: true })
+      if (!existsSync(resolved.id)) {
```
Comment (pi0) [inline packages/nitro/src/rollup/plugins/externals.ts:67]: rollup-plugin-resolve is preferred unless it does not resolve to full path

File: packages/nitro/src/rollup/config.ts [hunk 2]
```diff
@@ -273,7 +273,13 @@ export const getRollupConfig = (nitroContext: NitroContext) => {
         base: '/',
         processCwd: nitroContext._nuxt.rootDir,
         exportsOnly: true
-      }
+      },
+      exportConditions: [
```
Comment (danielroe) [inline packages/nitro/src/rollup/config.ts:277]: Should we pass these to node resolve too?

Comment (pi0) [reply packages/nitro/src/rollup/config.ts:277]: Yes. Actually we do in nitro config 😁

**Comment by pi0** [comment]
This change is rather risky. Doing it later with nitropack

**Pull request closed without merging**